### PR TITLE
Add extension service configuration to staging resources and remove extensions e2e test

### LINKED ISSUE #1593: Delete the extensions e2e test
After #1586, the extensions e2e test can be deleted.

Requires #1586

### PR DESCRIPTION
**Pull request by nikkiki**
Signed-off-by: Nikki Attea <nikki@sensu.io>

## What is this change?

Adds extension service configuration to staging resources and removes extensions e2e test

## Why is this change necessary?

Closes #1593 

## Does your change need a Changelog entry?

Yas.

## Do you need clarification on anything?

Nah.

## Were there any complications while making this change?

Nah.

## Have you reviewed and updated the documentation for this change? Is new documentation required?

Yes, the manual staging environment doc has been updated with reproducible steps for the extension service.

### CODE REVIEW

File: rpc/extension/example/main.go [hunk 1]
```diff
@@ -27,9 +29,11 @@ func LastCheckStatusZero(event *types.Event) (bool, error) {
 	return event.Check.Status == 0, nil
 }
 
+// FailingCheck logs which entity/check is failing
 func FailingCheck(event *types.Event, mutated []byte) error {
 	if !event.HasCheck() {
 		log.Print("event does not contain check")
```
[the remark this answers is not in the document]
Comment (nikkiki) [reply rpc/extension/example/main.go]: We log the check name already https://github.com/sensu/sensu-go/blob/3fc9a2b7aee0b5322afa2ece262543cf6e927de1/rpc/extension/example/main.go#L38. Do you mean the event entity name perhaps?

**Comment by nikkiki** [comment]
@echlebek We're just using the example extension found here: https://github.com/sensu/sensu-go/blob/master/rpc/extension/example/main.go.

File: rpc/extension/example/main.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (echlebek) [reply rpc/extension/example/main.go]: Ah I see, nevermind :)

**Comment by nikkiki** [comment]
These resources, when used in the staging environment helped us identify https://github.com/sensu/sensu-go/issues/1750. We won't deploy them until the issue is resolved, but they're probably 🆗 to be merged in.

**Pull request merged**